### PR TITLE
INS-664 "queue" of requests

### PR DESCRIPTION
**Pull request by aalekseev-ru**
**- What I did**
ad hoc, need more work in INS-666
**- How I did it**
mutex lock requests to the same object

### CODE REVIEW

File: logicrunner/logicrunner.go [hunk 1]
```diff
@@ -51,7 +49,7 @@ type LogicRunner struct {
 	MessageBus      core.MessageBus
 	machinePrefs    []core.MachineType
 	Cfg             *configuration.LogicRunner
-	context         map[Ref]ExecutionContext // if object exists, we are validating or executing it right now
+	execution       map[Ref]*ExecutionContext // if object exists, we are validating or executing it right now
 	contextMutex    sync.Mutex
```
Comment (ruz) [inline logicrunner/logicrunner.go]: didn't rename mutex to executionMutex

File: logicrunner/logicrunner.go [hunk 2]
```diff
@@ -70,7 +68,7 @@ func NewLogicRunner(cfg *configuration.LogicRunner) (*LogicRunner, error) {
 	res := LogicRunner{
 		ArtifactManager: nil,
 		Cfg:             cfg,
-		context:         make(map[Ref]ExecutionContext),
+		execution:       make(map[Ref]*ExecutionContext),
```
Comment (ruz) [inline logicrunner/logicrunner.go]: executionContext -> executionState

File: logicrunner/logicrunner.go [hunk 3]
```diff
@@ -308,10 +298,9 @@ func (lr *LogicRunner) getObjectMessage(objref Ref) (*objectBody, error) {
 }
 
 func (lr *LogicRunner) executeMethodCall(ctx core.LogicCallContext, m *message.CallMethod, vb ValidationBehaviour) (core.Reply, error) {
-	ec := ExecutionContext{}
-	if !lr.SetContext(m.ObjectRef, ec) {
-		return nil, errors.New("Method already executing")
-	}
+	data := lr.UpsertExecution(m.ObjectRef)
```
Comment (ruz) [inline logicrunner/logicrunner.go]: date -> es/state/or something similar

File: logicrunner/logicrunner.go [hunk 4]
```diff
@@ -375,10 +359,9 @@ func (lr *LogicRunner) executeMethodCall(ctx core.LogicCallContext, m *message.C
 }
 
 func (lr *LogicRunner) executeConstructorCall(ctx core.LogicCallContext, m *message.CallConstructor, vb ValidationBehaviour) (core.Reply, error) {
-	ec := ExecutionContext{}
-	if !lr.SetContext(m.GetRequest(), ec) {
-		return nil, errors.New("Constructor already executing by you")
-	}
+	executionState := lr.UpsertExecution(m.GetRequest())
```
Comment (Vany) [inline logicrunner/logicrunner.go]:
```suggestion
	executionState := lr.UpsertExecution(m.GetReference())
```

**Pull request merged**